File: src/types/tenantAdminShareProfile.ts
export interface TenantAdminShareProfile {
  storeDisplayName: string
  shareDisplayName: string
  shareAddress: string
  googleMapUrl: string
  shareContactPhone: string
  whatsappBusinessPhoneE164: string
  reservationShareNote: string
  reservationShareTemplate: string
  defaultReservationShareTemplate: string
  availableVariables: string[]
  usesDefaultReservationShareTemplate: boolean
}

export interface TenantAdminShareProfileMutation {
  shareDisplayName?: string | null
  googleMapUrl?: string | null
  whatsappBusinessPhoneE164?: string | null
  reservationShareNote?: string | null
  reservationShareTemplate?: string | null
}

export interface TenantAdminShareTemplateMutation {
  reservationShareTemplate?: string | null
}

export interface TenantAdminShareProfileResponse {
  success: true
  shareProfile: TenantAdminShareProfile
}

export interface TenantAdminSharePreviewResponse {
  success: true
  preview: {
    shareText: string
  }
}

File: src/main/java/com/rpb/reservation/tenantadmin/application/TenantAdminShareProfile.java
package com.rpb.reservation.tenantadmin.application;

import java.util.List;

public record TenantAdminShareProfile(
    String storeDisplayName,
    String shareDisplayName,
    String shareAddress,
    String googleMapUrl,
    String shareContactPhone,
    String whatsappBusinessPhoneE164,
    String reservationShareNote,
    String reservationShareTemplate,
    String defaultReservationShareTemplate,
    List<String> availableVariables,
    boolean usesDefaultReservationShareTemplate
) {
    public TenantAdminShareProfile {
        availableVariables = availableVariables == null ? List.of() : List.copyOf(availableVariables);
    }
}
